bin/index.js

```diff
@@ -1,71 +1,561 @@
+  }
+}
+
+async function rebootFunc(argv) {
+
+  let playerUser = players[argv.playerName].username;
+  let playerIP = players[argv.playerName].ipAddress;
+  let playerPW = players[argv.playerName].password;
+
+  let requestOptions = {
+    method: 'PUT',
+    url: 'http://' + playerIP + '/api/v1/control/reboot',
+  };
+
+  try {
+    let response = await requestFetch(requestOptions);
+    //console.log(response);
+    console.log('Player rebooted: ' + response.data.result.success);
+  } catch (err) {
+    console.log(err);
+  }
+
+}
+
+function addPlayerFunc(argv) {
+  fs.readFile('./bin/players.json', 'utf8', (error, data) => {
+    if (error) {
+      console.error('Error reading file: ', error);
+      return;
     }
-    return response.data?.data.result || response.data.result;
-};
 
-async function main() {
+    // parse json object
+    let JSONdata = JSON.parse(data);
+
+    // add new player
+    JSONdata[argv.playerName] = {
+      ipAddress: argv.ipAddress,
+      password: argv.password,
+      username: argv.username
+    }
+
+    // write new json object to file
+    let modifiedData = JSON.stringify(JSONdata, null, 2);
+    fs.writeFile('./bin/players.json', modifiedData, 'utf8', (error) => {
+      if (error) {
+        console.error('Error writing file: ', error);
+        return;
+      }
+      console.log('Player added successfully');
+    });
+  });
+}
+
+function removePlayerFunc(argv) {
+  fs.readFile('./bin/players.json', 'utf8', (error, data) => {
+    if (error) {
+      console.error('Error reading file: ', error);
+      return;
+    }
+
+    // parse json object
+    let JSONdata = JSON.parse(data);
+
+    // remove player
+    delete JSONdata[argv.playerName];
+
+    // write new json object to file
+    let modifiedData = JSON.stringify(JSONdata, null, 2);
+    fs.writeFile('./bin/players.json', modifiedData, 'utf8', (error) => {
+      if (error) {
+        console.error('Error writing file: ', error);
+        return;
+      }
+      console.log('Player removed successfully');
+    });
+  });
+}
+
+async function getDeviceInfo(argv) {
+
+  let playerUser = players[argv.playerName].username;
+  let playerIP = players[argv.playerName].ipAddress;
+  let playerPW = players[argv.playerName].password;
+
+  let requestOptions = {
+    method: 'GET',
+    url: 'http://' + playerIP + '/api/v1/info',
+  };
   try {
-    let response = await requestAxios({method: `${requestMethod}`, url: `${requestRoute}`});
-    // console.log(JSON.stringify(response));
-  } catch (e) {
-    console.log(JSON.stringify(e));
+    let response = await requestFetch(requestOptions);
+    console.log(response);
+  } catch (err) {
+    console.log(err);
   }
+  
+
 }
 
-if (require.main === module) {
-	main();
-}
\ No newline at end of file
+async function screenshotFunc(argv) {
+
+  let playerUser = players[argv.playerName].username;
+  let playerIP = players[argv.playerName].ipAddress;
+  let playerPW = players[argv.playerName].password;
```

Comment:
nit: consideration to include a quick helper function to grab the common "details" from the argv obj.

Reply:
smart - will include this in a future versions